docs/how_to/sessionize.md

```diff
@@ -0,0 +1,72 @@
+# How to Sessionize a Log of Events
+
+Suppose you have entities (users, objects, actions, etc) that have event logs through polling or event triggers.
+
+You might be interested in partitioning these logs by something called **sessions**, which can be defined as the duration of an event.
+
+In the case of a user portal, it might be the time spent completing a task or navigating an app.
+For games, it might be a time spent playing the game or remaining logged in.
+For retail, it might be checking out or walking the premises.
+
+This guide on sessionization is inspired by [_The Expressions API in Polars is Amazing_](https://www.pola.rs/posts/the-expressions-api-in-polars-is-amazing/),
+a blog post in the [Polars](https://www.pola.rs/) community demonstrating the strength of polars expressions.
+
+## Sessionizing Logs on a Cadence
+
+For this example, we have a dataset that contains entities polled on a cadence.
+
+Our data contains the following:
+
+- `char` : a unique identifier for a character (or a player). This is our `entity_col`
+- `timestamp`: a timestamp denoting when a `char` was polled. This occurs every ~10 minutes
+
+We can take this information, along with a definition of what separates two sessions for an entity, and break our dataset up into sessions **without using any joins**:
+
+```python
+# Given:
+# * `entity_col`: name of a column noting an entity identifier, e.g. an event or user
+# * `timestamp_col`: name of a column noting the timestamp for a row, e.g. when entity pingged
+# * `session_boundary_threshold`: integer delay in seconds noting if a row should be included
+#   in the previous session for an entity
+#   * i.e. if two adjacent rows for an entity are >n seconds apart then they are in two separate sessions
+
+# Window for finding session ids per character
+win_lag = ibis.cumulative_window(group_by=entity_col, order_by=timestamp_col)
+
+# Window for finding session min/max
+win_dur = ibis.window(group_by=[entity_col, 'session_id'])
+
+# Take the previous timestamp within a window (by charcter ordered by timestamp):
+# Note: the first value in a window will be null
+ts_lag = data.timestamp.lag().over(win_lag)
+
+# Subtract the lag from the current timestamp to get a timedelta
+ts_delta = data.timestamp - ts_lag
+
+# Compare timedelta to our session delay in seconds to determine if the
+# current timestamp falls outside of the session.
+is_new_session = ts_delta > ibis.interval(seconds=session_boundary_threshold)
+
+# Generate all of the data we need to analyze sessions:
+result = (
+    data
+    # Use our boolean column to determine whether a row falls outside of the current session
+    # fillna (coalesce) True to mark a new character, new sessions with True
+    # cast value to integer to mark True as 1 and False a 0 (for agg in next step)
+    .mutate(
+        # Include `ts_delta` in case it's needed after execution
+        ts_delta=ts_delta.fillna(ibis.interval(seconds=0)),
```

Comment:
Sounds like a YAGNI to me :)

Reply:
hah! It definitely is; users should be able to figure out if they want it they can just grab it.

Reply:
Done; thanks!